### PR TITLE
Complain on snapshot operations if unsupported

### LINKED ISSUE #3328: Snapshot operations do not fail when snapshots are unsupported
```
$ multipass ls
Name                    State             IPv4             Image
becoming-pike           Running           N/A              Ubuntu 23.04
luminous-merganser      Running           N/A              Ubuntu 20.04 LTS
nurturing-booklouse     Running           N/A              Ubuntu 22.04 LTS
valued-sheepshead       Running           N/A              Ubuntu 20.04 LTS
$ multipass list --snapshots
No snapshots found.
$ multipass info --snapshots
No snapshots found.
$ multipass restore luminous-merganser.wrong
restore failed: Multipass can only restore snapshots of stopped instances.
$ multipass stop luminous-merganser
$ multipass restore luminous-merganser.wrong
restore failed: No such snapshot: luminous-merganser.wrong
$ multipass snapshot luminous-merganser
snapshot failed: The Snapshots feature is not implemented on this backend.
```

### PR DESCRIPTION
**Pull request by ricab**
And omit the number of snapshots in `info`.

Fixes #3328

### CODE REVIEW

File: src/platform/backends/shared/base_virtual_machine.h [hunk 1]
```diff
@@ -129,15 +130,22 @@ class BaseVirtualMachine : public VirtualMachine
...
 inline int multipass::BaseVirtualMachine::get_snapshot_count() const
 {
+    require_snapshots_support();
     const std::unique_lock lock{snapshot_mutex};
     return snapshot_count;
 }
 
+inline void multipass::BaseVirtualMachine::require_snapshots_support() const
+{
+    throw NotImplementedOnThisBackendException{"Snapshots"};
```
Comment (townsend2010) [inline src/platform/backends/shared/base_virtual_machine.h:148]: I think this should be lower case `snapshots`.  Making it lower case is also in line with `networks` and the like.

File: src/platform/backends/shared/base_virtual_machine_factory.h [hunk 2]
```diff
@@ -102,4 +104,9 @@ inline void multipass::BaseVirtualMachineFactory::remove_resources_for(const std
     instance_dir.removeRecursively();
 }
 
+inline void multipass::BaseVirtualMachineFactory::require_snapshots_support() const
+{
+    throw NotImplementedOnThisBackendException{"Snapshots"};
```
Comment (townsend2010) [inline src/platform/backends/shared/base_virtual_machine_factory.h:109]: idem, lower case `snapshots`.

**Review by townsend2010 — CHANGES_REQUESTED** [review]
Hey @ricab!

Thanks for this! Works well and the code looks good.  I just have a couple of consistency comments below.

File: src/platform/backends/shared/base_virtual_machine.h (hunk 1 above)
Comment (ricab) [reply src/platform/backends/shared/base_virtual_machine.h]: Good point. Done.

File: src/platform/backends/shared/base_virtual_machine_factory.h (hunk 2 above)
Comment (ricab) [reply src/platform/backends/shared/base_virtual_machine_factory.h]: Done.

**Review by townsend2010 — APPROVED** [review]
Ok, LGTM, thanks!

**Comment by ricab** [comment]
I found an issue...

**Comment by ricab** [comment]
Fixed. The issue was in snapshot settings, for example:

```
$ multipass get --keys
get failed: The snapshots feature is not implemented on this backend.
```

**Pull request merged**